src/Raven.Client/Documents/Session/DocumentSessionAttachmentsBase.cs

```diff
@@ -71,20 +71,30 @@ public void Store(string documentId, string name, Stream stream, string contentT
         public void Store(object entity, string name, Stream stream, string contentType = null)
         {
             if (DocumentsByEntity.TryGetValue(entity, out var document) == false)
-                ThrowEntityNotInSession(entity);
+                ThrowEntityNotInSessionOrMissingId(entity);
 
             Store(document.Id, name, stream, contentType);
         }
 
-        protected void ThrowEntityNotInSession(object entity)
+        protected void ThrowEntityNotInSessionOrMissingId(object entity)
         {
             throw new ArgumentException($"{entity} is not associated with the session. Use documentId instead or track the entity in the session.", nameof(entity));
         }
 
+        protected void ThrowEntityNotInSession(object entity)
```

Comment:
I do not like it, create 2 overloads

Reply:
I cannot overload with an object,
pulled out the if to implementation of getNames